src/test/scala/is/hail/TestUtils.scala

```diff
@@ -23,11 +23,12 @@ object TestUtils {
   def interceptException[E <: Throwable : Manifest](regex: String)(f: => Any) {
     val thrown = intercept[E](f)
     val p = regex.r.findFirstIn(thrown.getMessage).isDefined
+    val msg =
+      s"""expected fatal exception with pattern `$regex'
+         |  Found: ${ thrown.getMessage } """
     if (!p)
-      println(
-        s"""expected fatal exception with pattern `$regex'
-           |  Found: ${ thrown.getMessage } """.stripMargin)
-    assert(p)
+      println(msg)
+    assert(p, msg)
```

Comment:
Doesn't assert print the message if `p` is false?

Reply:
what was happening was that the message got printed to the stdout, but I'm using SBT to run tests (it re-compiles the relevant files and runs the test suites for this PR in 6 seconds, so it's really nice) and SBT shows me a short summary of what went wrong with only the assert message. So it was really frustrating to see:
```
[info] - bothSidesSliceFunctionOutOfBoundsNotFatal *** FAILED ***
[info]   is.hail.utils.AssertionError: p was false
```
and then I had to scroll up to see the output.

This makes the message something like:

```
[info] - bothSidesSliceFunctionOutOfBoundsNotFatal *** FAILED ***
[info]   t.valuesSimilar(c, expected, t.valuesSimilar$default$3, t.valuesSimilar$default$4) was false  , ab (TestUtils.scala:419)
```

which still isn't great but it at least has information about the actual value and the expected value.

Reply:
Ideally the message would be 
```
[info] - bothSidesSliceFunctionOutOfBoundsNotFatal *** FAILED ***
[info]   expected "��" to equal "ab" (TestUtils.scala:419)
```
But I wasn't sure how to make assert not show that useless extra information (I didn't look very hard either though).